File: src/main/java/thermos/ThermosRemapper.java
package thermos;

import net.md_5.specialsource.JarMapping;
import net.md_5.specialsource.JarRemapper;

public class ThermosRemapper extends JarRemapper {
    public ThermosRemapper(JarMapping jarMapping) {
        super(jarMapping);
    }

    @Override
    public String mapSignature(final String signature, final boolean typeSignature) {
        try {
            return super.mapSignature(signature, typeSignature);
        } catch (Exception e) {
            return signature;
        }
    }

    @Override
    public String mapFieldName(final String owner, final String name, final String desc, final int access) {
        return super.mapFieldName(owner, name, desc, -1);
    }
}

File: src/main/kotlin/kettle/ThermosRemapper.kt
package kettle

import net.md_5.specialsource.JarMapping
import net.md_5.specialsource.JarRemapper

class ThermosRemapper(jarMapping: JarMapping) : JarRemapper(jarMapping) {

    override fun mapSignature(signature: String, typeSignature: Boolean): String {
        try {
            return super.mapSignature(signature, typeSignature)
        } catch (e: Exception) {
            return signature
        }

    }
}
